### PR TITLE
[Package Signing] Improve resiliency against null pointers

### PR DESCRIPTION
**Pull request by loic-sharma**
See this issue for more information: https://github.com/NuGet/Engineering/issues/2649

/cc @dtivel

### CODE REVIEW

File: src/Validation.PackageSigning.ValidateCertificate/CertificateValidationService.cs [hunk 1]
```diff
@@ -58,14 +58,14 @@ public async Task<bool> TrySaveResultAsync(EndCertificateValidation validation,
                 _logger.LogWarning(
                     "Updating previously revoked certificate {CertificateThumbprint} to status {NewStatus}",
                     validation.EndCertificate.Thumbprint,
-                    result.Status);
+                    result);
```
Comment (loic-sharma) [inline src/Validation.PackageSigning.ValidateCertificate/CertificateValidationService.cs]: This object has a `ToString()` method that will provide more information

File: src/Validation.PackageSigning.ValidateCertificate/OnlineCertificateVerifier.cs [hunk 2]
```diff
@@ -233,7 +239,9 @@ private unsafe bool CertificateWasVerifiedOnline(CERT_REVOCATION_INFO* pRevocati
 
         private unsafe DateTime? GetRevocationTime(CERT_REVOCATION_INFO* pRevocationInfo)
         {
-            if (pRevocationInfo->dwRevocationResult == CertTrustErrorStatus.CERT_TRUST_NO_ERROR)
+            if (pRevocationInfo->dwRevocationResult == CertTrustErrorStatus.CERT_TRUST_NO_ERROR
+                || pRevocationInfo->pCrlInfo == null
+                || pRevocationInfo->pCrlInfo->pCrlEntry == null)
```
Comment (loic-sharma) [inline src/Validation.PackageSigning.ValidateCertificate/OnlineCertificateVerifier.cs]: This `pCrlEntry` was `null`, thereby causing the null reference exception. Lines 179-182 should prevent this case from happening.

File: src/Validation.PackageSigning.ValidateCertificate/OnlineCertificateVerifier.cs [hunk 3]
```diff
@@ -233,7 +239,9 @@ private unsafe bool CertificateWasVerifiedOnline(CERT_REVOCATION_INFO* pRevocati
 
         private unsafe DateTime? GetRevocationTime(CERT_REVOCATION_INFO* pRevocationInfo)
         {
-            if (pRevocationInfo->dwRevocationResult == CertTrustErrorStatus.CERT_TRUST_NO_ERROR)
+            if (pRevocationInfo->dwRevocationResult == CertTrustErrorStatus.CERT_TRUST_NO_ERROR
```
Comment (joelverhagen) [inline src/Validation.PackageSigning.ValidateCertificate/OnlineCertificateVerifier.cs]: Do we need similar null checks in `GetStatusUpdateTime`?

File: src/Validation.PackageSigning.ValidateCertificate/OnlineCertificateVerifier.cs [hunk 4]
```diff
@@ -175,6 +175,12 @@ private unsafe void InspectChain(X509Chain chain, X509Certificate2 certificate,
             var addedRef = false;
             var chainHandle = chain.SafeHandle;
 
+            // Don't bother trying to get revocation information if the revocation status is unknown.
+            if (chain.ChainStatus.Any(s => s.Status == X509ChainStatusFlags.RevocationStatusUnknown))
```
Comment (joelverhagen) [inline src/Validation.PackageSigning.ValidateCertificate/OnlineCertificateVerifier.cs]: Maybe i'm missing something, but it looks like we're doing two changes here for one null ref. What does adding this `RevocationStatusUnknown` add to the picture in addition to the null checks?

File: src/Validation.PackageSigning.ValidateCertificate/OnlineCertificateVerifier.cs (hunk 3 above)
Comment (loic-sharma) [reply src/Validation.PackageSigning.ValidateCertificate/OnlineCertificateVerifier.cs]: Added (this is already covered by the method that calls `GetStatusUpdateTime`, but I'd rather err on safety)

**Pull request merged**